lib/Doctrine/DBAL/Driver/OCI8/OCI8Statement.php

```diff
@@ -134,7 +141,14 @@ static public function convertPositionalToNamedPlaceholders($statement)
      */
     public function bindValue($param, $value, $type = null)
     {
-        return $this->bindParam($param, $value, $type, null);
+        if ($value instanceof OCI8Descriptor) {
```

Comment:
> This is somewhat against the Statement::bindValue() contract which says that $value is the actual value to be bound. I see that OCI8Descriptor "represents" this particular value but this will break the usage of the DBAL OCI8 driver as standalone component. 

Indeed, that's the solution I used because the `Type::getBindingType` method isn't platform aware. If it was, we could easily give a `PDO::PARAM_CLOB` or `PDO::PARAM_LOB`.  

> Furthermore in such case the $type parameter is not declaring anymore HOW the value is to be bound (because it is ignored). How should the user know how to properly use this API? It does not conform to the given contract anymore.

Right, the type is here `OCI8Descriptor::$_type` (`OCI_BLOB` or `OCI_CLOB`). Still, the default PDO behavior is unchanged. For example, `->bindValue(':param', $blobdescriptor, OCI_BLOB)` is a valid call but  `->bindValue(':param', $blobdescriptor, \PDO::PARAM_LOB)` is not with this implementation (can be fixed).

> I still wonder why we need to use descriptors for CLOB bindings at all. Can't we simply use the value + \PDO::PARAM_STR? Is there a failing test case?

Because, that's how Oracle stores big text fields. We could change the `TextType` to give a `varchar2` with oracle but it would be length-restricted.
For a failing test case, see the `DBAL2386Test` below. It will fail on the #2262 PR because of CLOBS.

>  If that would be the case how does the driver determine whether a descriptor is necessary or not?

If it's a BLOB or a CLOB the driver wants a descriptor.

> If we cannot use PARAM_STR without descriptors at all anymore (which would be really weird)

No issues with this, it's was left unchanged for `StringType` for example.

> \PDO::PARAM_LOB should then as already said always use a BLOB descriptor.

Indeed, I may add a fix for this.